### DISCUSSION TITLE
Add routes only based on current locale used

### DISCUSSION THREAD

**Discussion by szwenni** [nuxt-modules/i18n#2668] [Q&A]
### Describe the feature

Our project right now contains several rules and several languages. With this increased complexity the "entry" bundle for nuxt consists 90% of route definition for vue-router. The parsing of this whole entries costs a lot of main thread work which is not needed in this place of time. I think for initial loading the page only the route definitions of the current language and the ones for all languages of the current page are needed. The others can be lazy loaded based on laguage selection if I'm not wrong.

I know this requires some effort to implement but I think nuxt/vue provides a way of adding and removing routes at runtime. I'm also not quotes sure right now where the correct way to implement this would be and how this can be achieved at all.
I'm also not quite sure if e.g. splitting away the routes from the entry file can affect this in a positive way.

I'm willing to help if someone can give me a good start point where to start.

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution) (The contribution guideline of nuxt-modules/i18n is compliant with Nuxt too).
- [X] Check existing [discussions](https://github.com/nuxt-modules/i18n/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by kazupon** [comment]
Thank you for your feedback.

Your issue is not a bug and will be moved to github dicussions.

And to solve your issue, we need more details information (e.g. your minimal project or code).
